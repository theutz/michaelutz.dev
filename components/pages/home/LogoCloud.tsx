/* eslint-disable @next/next/no-img-element */
import Image, { ImageProps } from "next/image";
import unityLogo from "public/assets/images/unity-logo.png";
import smartlyLogo from "public/assets/images/smartly-logo.png";
import quenticLogo from "public/assets/images/quentic-logo.png";
import uhsLogo from "public/assets/images/uhs-logo.png";
import ersLogo from "public/assets/images/ers-logo.png";
import { classNames } from "utils/classNames";

export const companies: readonly Pick<ItemProps, "src" | "alt">[] = [
  { src: unityLogo, alt: "Unity" },
  { src: smartlyLogo, alt: "Smartly.io" },
  { src: quenticLogo, alt: "Quentic" },
  { src: uhsLogo, alt: "Universal Hospital Systems" },
  { src: ersLogo, alt: "Emergency Reporting" },
];

export function LogoCloud() {
  return (
    <div className="bg-gray-100" data-cy="logo-cloud">
      <div className="px-4 py-16 mx-auto max-w-7xl sm:px-6 lg:px-8">
        <p className="text-sm font-semibold tracking-wide text-center text-gray-500 uppercase">
          Trusted by a handful of former employers
        </p>
        <div
          data-cy="logo-cloud-container"
          className="mt-6 grid grid-cols-2 gap-8 md:grid-cols-6 lg:grid-cols-5"
        >
          {companies.map((company, i) => (
            <Item
              key={company.alt}
              {...company}
              current={i + 1}
              length={companies.length}
            />
          ))}
        </div>
      </div>
    </div>
  );
}

type ItemProps = {
  src: Exclude<ImageProps["src"], string>;
  alt: ImageProps["alt"];
  current: number;
  length: number;
};

function Item({ src, alt, current, length }: ItemProps) {
  const isLast = current === length;
  const is2ndToLast = length - 1 === current;

  return (
    <div
      className={classNames(
        "relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1",
        isLast ? "col-span-2 md:col-start-4" : "col-span-1",
        is2ndToLast && "md:col-start-2"
      )}
    >
      <Image layout="fill" objectFit="contain" src={src} alt={alt} />
    </div>
  );
}

export default LogoCloud;
