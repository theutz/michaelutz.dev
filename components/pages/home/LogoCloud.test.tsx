import { render, screen } from "@testing-library/react"
import { LogoCloud } from "./LogoCloud"

describe("<LogoCloud />", () => {
  it("renders", () => {
    // Arrange

    // Act
    const { asFragment } = render(<LogoCloud />)

    // Assert
    expect(asFragment()).toMatchSnapshot()
  })
})
