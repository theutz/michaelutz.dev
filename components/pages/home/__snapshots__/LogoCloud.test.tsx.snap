// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoCloud /> renders 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-100"
  >
    <div
      class="px-4 py-16 mx-auto max-w-7xl sm:px-6 lg:px-8"
    >
      <p
        class="text-sm font-semibold tracking-wide text-center text-gray-500 uppercase"
      >
        Trusted by a handful of former employers
      </p>
      <div
        class="mt-6 grid grid-cols-2 gap-8 md:grid-cols-6 lg:grid-cols-5"
      >
        <div
          class="relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1 col-span-1"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <noscript />
            <img
              alt="Unity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
          </div>
        </div>
        <div
          class="relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1 col-span-1"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <noscript />
            <img
              alt="Smartly.io"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
          </div>
        </div>
        <div
          class="relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1 col-span-1"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <noscript />
            <img
              alt="Quentic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
          </div>
        </div>
        <div
          class="relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1 col-span-1 md:col-start-2"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <noscript />
            <img
              alt="Universal Hospital Systems"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
          </div>
        </div>
        <div
          class="relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1 col-span-2 md:col-start-4"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <noscript />
            <img
              alt="Emergency Reporting"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
